tools/sloths/src/components/memory/GameField.vue

```diff
@@ -0,0 +1,468 @@
+        `${MEMORY_GAME_SRPITE}#memory11`,
+        `${MEMORY_GAME_SRPITE}#memory12`,
+      ];
+
+      this.getCards();
+    },
+
+    getCards() {
+      this.changeScrollHidden('hidden');
+      this.cards = [];
+      let index = 0;
+
+      const images = this.images.sort(() => Math.random() - 0.5).filter((el, i) => i < this.level.n);
+
+      images.forEach((el, i) => {
+        this.cards.push({ img: el, id: i, index, open: false, success: false });
+        index += 1;
+        this.cards.push({ img: el, id: i, index, open: false, success: false });
+        index += 1;
+      });
+      setTimeout(() => this.changeScrollHidden(''), MEMORY_GAME_TIMEOUT);
+    },
+
+    changeScrollHidden(val = '') {
+      const mainEl: HTMLElement | null = document.querySelector('.main');
+      if (mainEl instanceof HTMLElement) {
+        mainEl.style.overflowY = val;
+      }
+    },
+
+    startGame() {
+      const isAllClosed = this.cards.every((el) => !el.open);
+
+      if (!isAllClosed) {
+        playAudio(audioFlip);
+        this.cards.forEach((el, i) => this.closeCard(i));
+      }
+
+      this.activeCard = Infinity;
+      this.steps = 0;
+      this.startTime = 0;
+      this.endTime = 0;
+
+      if (isAllClosed) {
+        playAudio(audioSlide);
+        this.cards.sort(() => Math.random() - 0.5);
+      } else {
+        setTimeout(() => {
+          playAudio(audioSlide);
+          this.cards.sort(() => Math.random() - 0.5);
+        }, MEMORY_GAME_TIMEOUT);
+      }
+      setTimeout(() => {
+        playAudio(audioSlide);
+        this.cards.sort(() => Math.random() - 0.5);
+      }, MEMORY_GAME_TIMEOUT / 2);
+    },
+
+    getImage(i: number): string {
+      return this.cards[i].img;
+    },
+
+    getIsOpen(i: number): boolean {
+      return this.cards[i].open;
+    },
+
+    checkGameHandler(i: number): boolean {
+      return !(this.cards[i].open || this.isHandled || this.isAnimated);
+    },
+
+    cardsNotMatched(i1: number, i2: number) {
+      this.isHandled = true;
+
+      setTimeout(() => {
+        playAudio(audioFail);
+
+        this.closeCard(i1);
+        this.closeCard(i2);
+
+        this.isHandled = false;
+      }, MEMORY_GAME_TIMEOUT);
+    },
+
+    cardsMatched(i1: number, i2: number) {
+      this.isHandled = true;
+
+      setTimeout(() => {
+        playAudio(audioSuccess);
+
+        this.cards[i1].success = true;
+        this.cards[i2].success = true;
+
+        this.isHandled = false;
+      }, MEMORY_GAME_TIMEOUT);
+    },
+
+    gameHandler(i: number) {
+      if (this.startTime === 0) this.startTime = Date.now();
+
+      if (!this.checkGameHandler(i)) return;
+
+      playAudio(audioFlip);
+      this.openCard(i);
+      this.steps += 1;
+
+      if (this.activeCard === Infinity) {
+        this.activeCard = i;
+      } else if (this.cards[i].id === this.cards[this.activeCard].id) {
+        const i2 = this.activeCard;
+        this.activeCard = Infinity;
```

Comment:
Please consider to avoid duplication of these 2 lines

Reply:
done
add changeActiveCard() method